package_nso_to_oc/xe/xe_acls.py

```diff
@@ -0,0 +1,476 @@
+    for interface in acl_interfaces.values():
+        oc_acl_interface.append(interface)
+
+    process_ntp(config_before, config_after)
+    process_line(config_before, config_after)
+
+class BaseAcl:
+    def __init__(self, oc_acl_set, xe_acl_set, xe_acl_set_after):
+        self._oc_acl_set = oc_acl_set
+        self._xe_acl_set = xe_acl_set
+        self._xe_acl_set_after = xe_acl_set_after
+
+    def process_acl(self):
+        acl_set = {
+            "openconfig-acl:name": self._xe_acl_set.get("name"),
+            "openconfig-acl:type": self._acl_type,
+            "openconfig-acl:config": {
+                "openconfig-acl:name": self._xe_acl_set.get("name"),
+                "openconfig-acl:type": self._acl_type,
+                "openconfig-acl:description": self._xe_acl_set.get("name"), # XE doesn't seem to have a description.
+            },
+            "openconfig-acl:acl-entries": {
+                "openconfig-acl:acl-entry": []
+            }
+        }
+        self._oc_acl_set.append(acl_set)
+        del self._xe_acl_set_after["name"]
+        
+        for rule_index, access_rule in enumerate(self._xe_acl_set.get(self._rule_list_key, [])):
+            self.__set_rule_parts(access_rule, acl_set)
+            self._xe_acl_set_after[self._rule_list_key][rule_index] = None
+
+    def __set_rule_parts(self, access_rule, acl_set):
+        rule_parts = access_rule.get("rule", "").split()
+        
+        if len(rule_parts) < 1:
+            return
+
+        seq_id = int(rule_parts[0])
+        entry = {
+            "openconfig-acl:sequence-id": seq_id,
+            "openconfig-acl:config": {"openconfig-acl:sequence-id": seq_id},
+            "openconfig-acl:actions": {
+                "openconfig-acl:config": {"openconfig-acl:forwarding-action": actions_xe_to_oc[rule_parts[1]]}
+            }
+        }
+        current_index = self.__set_protocol(entry, rule_parts)
+        # Source IP
+        current_index = self.__set_ip_and_port(rule_parts, current_index, entry, True)
+        
+        if self._acl_type == "ACL_IPV4":
+            # Destination IP (if exists)
+            current_index = self.__set_ip_and_port(rule_parts, current_index, entry, False)
+
+        if len(rule_parts) > current_index and rule_parts[current_index] == "log-input":
+            entry["openconfig-acl:actions"]["openconfig-acl:config"]["openconfig-acl:log-action"] = "LOG_SYSLOG"
+        else:
+            entry["openconfig-acl:actions"]["openconfig-acl:config"]["openconfig-acl:log-action"] = "LOG_NONE"
+
+        acl_set["openconfig-acl:acl-entries"]["openconfig-acl:acl-entry"].append(entry)
+    
+    def __set_protocol(self, entry, rule_parts):
+        if self._acl_type == "ACL_IPV4_STANDARD":
+            return 2
+        if rule_parts[2] != 'ip':
+            self.__get_ipv4_config(entry)["openconfig-acl:protocol"] = protocols_oc_to_xe[rule_parts[2]]
+        
```

Comment:
If protocol is not in protocols_oc_to_xe, we need to log, continue processing access-list statement but not add ACL to OpenConfig ACLs.